profiles/views.py

```diff
@@ -174,6 +176,16 @@ def get_serializer_class(self):
         else:
             return ProfileOwnerDetailEditSerializer
 
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        user = self.request.user
+        password = self.request.data.get("password")
+        if not password or not check_password(password, user.password):
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
1. `400 BAD REQUEST` must have a body, with explanation what exactly is wrong with the request. So, I guess this logic (together with password check) should be in serializer validator.
2. Don't, don't do `return Response()`.

Reply:
I wanted to do it in serializer, but there is no delete method in serializers, so I don't know how to implement it. I remember that it's not okay to return Response. But I just override original destroy method and it contains Response.